plugins/out_cef/cef.c

```diff
@@ -0,0 +1,1150 @@
+        struct tm tm;
+        char time_formatted[64];
+        int severity = 6; /* INFO */
+        int facility = 1; /* USER */
+        unsigned int pri;
+
+        if (msg->syslog_facility >= 0) {
+            facility = msg->syslog_facility;
+        }
+
+        if (msg->syslog_severity >= 0) {
+            severity = msg->syslog_severity;
+        }
+
+        pri = (facility << 3) + severity;
+
+        tmp = flb_sds_printf(s, "<%i>", pri);
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+
+        gmtime_r(&tms->tm.tv_sec, &tm);
+
+        ret = strftime(time_formatted, sizeof(time_formatted) - 1,
+                       "%b %d %H:%M:%S ", &tm);
+        if (ret <= 0) {
+          return NULL;
+        }
+
+        tmp = flb_sds_cat(*s, time_formatted, ret);
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+
+        if (msg->host)
+        {
+            tmp = flb_sds_cat(*s, msg->host, flb_sds_len(msg->host));
+            if (!tmp) {
+              return NULL;
+            }
+            *s = tmp;
+        }
+        else {
+            tmp = flb_sds_cat(*s, cef_hostname, cef_hostname_len);
+            if (!tmp) {
+              return NULL;
+            }
+            *s = tmp;
+        }
+        tmp = flb_sds_cat(*s, " ", 1);
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+    }
+
+    tmp = flb_sds_cat(*s, "CEF:0|", 6);
+    if (!tmp) {
+      return NULL;
+    }
+    *s = tmp;
+
+    if (msg->dev_vendor)
+    {
+        tmp = flb_sds_cat_utf8(s, msg->dev_vendor, flb_sds_len(msg->dev_vendor),
+                               esc_cef_msg, sizeof(esc_cef_msg));
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+    }
+
+    tmp = flb_sds_cat(*s, "|", 1);
+    if (!tmp) {
+      return NULL;
+    }
+    *s = tmp;
+
+    if (msg->dev_product)
+    {
+        tmp = flb_sds_cat_utf8(s, msg->dev_product, flb_sds_len(msg->dev_product),
+                               esc_cef_msg, sizeof(esc_cef_msg));
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+    }
+
+    tmp = flb_sds_cat(*s, "|", 1);
+    if (!tmp) {
+      return NULL;
+    }
+    *s = tmp;
+
+    if (msg->dev_version)
+    {
+        tmp = flb_sds_cat_utf8(s, msg->dev_version, flb_sds_len(msg->dev_version),
+                               esc_cef_msg, sizeof(esc_cef_msg));
+        if (!tmp) {
+          return NULL;
+        }
+        *s = tmp;
+    }
+
+    tmp = flb_sds_cat(*s, "|", 1);
+    if (!tmp) {
+      return NULL;
+    }
+    *s = tmp;
+
+    if (msg->dev_event_cid)
+    {
+        tmp = flb_sds_cat_utf8(s, msg->dev_event_cid, flb_sds_len(msg->dev_event_cid),
+                               esc_cef_msg, sizeof(esc_cef_msg));
+        if (!tmp) {
+           return NULL;
+        }
+        *s = tmp;
+    }
+    tmp = flb_sds_cat(*s, "|", 1);
+    if (!tmp) {
+      return NULL;
+    }
+    *s = tmp;
+
+    if (msg->name)
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions